apps/asap-server/test/utils/sync-google-event.test.ts

```diff
@@ -26,12 +50,28 @@ describe('Sync calendar util hook', () => {
         endDateTimeZone: 'Europe/Lisbon',
         status: 'Confirmed',
         calendar: ['squidex-calendar-id'],
-        tags: [],
       },
     );
   });
 
-  test('Should upsert event - with dateTime', async () => {
+  test('Should throw when a remote operation throws', async () => {
+    eventControllerMock.fetchByGoogleId.mockRejectedValueOnce(
+      new Error('Squidex'),
+    );
+    await expect(syncEvent(event, defaultCalendarTimezone)).rejects.toThrow();
+
+    eventControllerMock.fetchByGoogleId.mockResolvedValueOnce(restEvent);
+    eventControllerMock.update.mockRejectedValueOnce(new Error('Squidex'));
+    await expect(syncEvent(event, defaultCalendarTimezone)).rejects.toThrow();
```

Comment:
I know its a lot of copy and paste but I think we should split this into 3 different tests, its not a HUGE deal but generally speaking a single test should only test one piece of logic (some people even say that there should only be one assertion per test but thats crazy)